pkgs/development/web/bun/default.nix

```diff
@@ -64,4 +64,20 @@ stdenvNoCC.mkDerivation rec {
     maintainers = with maintainers; [ DAlperin jk ];
     platforms = builtins.attrNames dists;
   };
+
+  updateScript = writeShellScript "update-bun" ''
+      set -o errexit
+      export PATH="${lib.makeBinPath [ curl jq common-updater-scripts ]}"
+      NEW_VERSION=$(curl -s https://api.github.com/oven-sh/bun/release | jq ".[1].tag_name" --raw-output)
+      FORMATTED_OLD="bun-v${version}"
+      if [[ "${FORMATTED_OLD}" = "$NEW_VERSION" ]]; then
+          echo "The new version same as the old version."
+          exit 0
+      fi
+      USABLE_NEW_VERSION = printf '%s\n' "${NEW_VERSION//bun-v/}"
+      for platform in ${lib.escapeShellArgs meta.platforms}; do
+        update-source-version "bun" "0" "${lib.fakeSha256}" --source-key="sources.$platform"
```

Comment:
You need to have `sources` attribute containing the per-platform sources for this to work. You can add it to passthru too.